### PR TITLE
[TGBOT-25] Create Engineer repository & Bean using Arrays, CRUD and Tests

### PR DESCRIPTION
**Pull request by schneidervs**
https://schneidervs.atlassian.net/browse/TGBOT-25

### CODE REVIEW

File: stoservice/src/main/java/com/tutrit/stoservice/repository/EngineerRepository.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    public void deleteEngineer(String a) {
+        int i = Integer.parseInt(a);
+        deleteEngineer(i);
+    }
+
+    public void deleteEngineer(Engineer engineer) {
+        Engineer find = extract(engineer);
+        int i = Integer.parseInt(findEngineer(find).getIdEngineer());
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/EngineerRepository.java:81]: Why are you extracting the engineer here if he is extracting in the findEngineer()?

File: stoservice/src/main/java/com/tutrit/stoservice/repository/EngineerRepository.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    public Engineer findEngineer(String a) {
+        int i = Integer.parseInt(a);
+        return findEngineer(i);
+    }
+
+    public Engineer findEngineer(Engineer engineer) {
+        Engineer find = extract(engineer);
+        for (Engineer e : engineers) {
+            extFind = extract(e);
+            if (find.equals(extFind)) {
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/EngineerRepository.java:49]: What is the need to extract the engineer if you can compare it directly in the if statement?

File: stoservice/src/main/java/com/tutrit/stoservice/repository/EngineerRepository.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    public Engineer findEngineer(int i) {
+        if (i > (engineers.length - 1)) {
+            log.info("Array out of bounds");
+            return new Engineer();
+        }
+        return engineers[i - 1];
+    }
+
+    public Engineer findEngineer(String a) {
+        int i = Integer.parseInt(a);
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/EngineerRepository.java:41]: Maybe put it directly in the return parameters of the method?

**Review by yurasava — COMMENTED** [review]
Looks good. But there are questions about the extract() method.

**Review by Sav4ykVladzislau — APPROVED** [review]

**Review by max-java — APPROVED** [review]
we could merge it so far as we closing the sprint

**Review by AlexanderYarokhovich — APPROVED** [review]
good

**Review by yurasava — APPROVED** [review]

**Pull request merged**